core/types/receipt.go

```diff
@@ -141,10 +141,6 @@ func (r *Receipt) EncodeRLP(w io.Writer) error {
 	if r.Type == LegacyTxType {
 		return rlp.Encode(w, data)
 	}
-	// It's an EIP-2718 typed TX receipt.
-	if r.Type != AccessListTxType {
```

Comment:
If we want to keep the old semantics, we should do `if r.Type != AccessListTxType && r.Type != DynamicFeeTxType` instead of deleting the clause altogether.

Reply:
I ended up deleting this clause because it seems rather pathological. The code *only* supports `LegacyTx`, `AccessListTx`, and `DynamicFeeTx`, so it's essentially checking if the tx type is support when it's unmarshalled. Seems unnecessary to do again in the receipt logic.